mypy/test/testcheck.py

```diff
@@ -173,33 +182,36 @@ def run_case_once(self, testcase: DataDrivenTestCase, incremental: int = 0) -> N
         a = normalize_error_messages(a)
 
         # Make sure error messages match
-        if incremental == 0:
+        if incremental_step == 0:
+            # Not incremental
             msg = 'Invalid type checker output ({}, line {})'
             output = testcase.output
-        elif incremental == 1:
+        elif incremental_step == 1:
             msg = 'Invalid type checker output in incremental, run 1 ({}, line {})'
             output = testcase.output
-        elif incremental == 2:
-            msg = 'Invalid type checker output in incremental, run 2 ({}, line {})'
-            output = testcase.output2
+        elif incremental_step > 1:
+            msg = ('Invalid type checker output in incremental, run {}'.format(incremental_step) +
+                   ' ({}, line {})')
+            output = testcase.output2.get(incremental_step, [])
         else:
             raise AssertionError()
 
         if output != a and self.update_data:
             update_testcase_output(testcase, a)
         assert_string_arrays_equal(output, a, msg.format(testcase.file, testcase.line))
 
-        if incremental and res:
+        if incremental_step and res:
             if options.follow_imports == 'normal' and testcase.output is None:
                 self.verify_cache(module_data, a, res.manager)
-            if incremental == 2:
+            if incremental_step > 1:
+                suffix = '' if incremental_step == 2 else str(incremental_step - 1)
                 self.check_module_equivalence(
-                    'rechecked',
-                    testcase.expected_rechecked_modules,
+                    'rechecked' + suffix,
```

Comment:
Adding the suffix makes for a slightly awkward error message, "Set of rechecked1 modules does not match expected set". Especially since this refers to the difference between rechecked[1] and rechecked[2].